internal/setup/yaml.go

```diff
@@ -628,41 +687,58 @@ func pathInfoToYAML(pi *PathInfo) (*yamlPath, error) {
 	return path, nil
 }
 
-// sliceToYAML converts a Slice object to a yamlSlice object.
-func sliceToYAML(s *Slice) (*yamlSlice, error) {
-	slice := &yamlSlice{
+func v1v2SliceToYAML(s *Slice) (*v1v2YAMLSlice, error) {
+	oldSlice := &v1v2YAMLSlice{
 		Contents:    make(map[string]*yamlPath, len(s.Contents)),
 		Mutate:      s.Scripts.Mutate,
-		V3Essential: make(map[string]*yamlEssential, len(s.Essential)),
+		V3Essential: map[string]*yamlEssential{},
 	}
 	for key, info := range s.Essential {
-		slice.V3Essential[key.String()] = &yamlEssential{Arch: yamlArch{info.Arch}}
+		oldSlice.V3Essential[key.String()] = &yamlEssential{Arch: yamlArch{info.Arch}}
 	}
 	for path, info := range s.Contents {
 		yamlPath, err := pathInfoToYAML(&info)
 		if err != nil {
 			return nil, err
 		}
-		slice.Contents[path] = yamlPath
+		oldSlice.Contents[path] = yamlPath
 	}
-	return slice, nil
+	return oldSlice, nil
 }
 
-// packageToYAML converts a Package object to a yamlPackage object.
-func packageToYAML(p *Package) (*yamlPackage, error) {
-	pkg := &yamlPackage{
-		Name:    p.Name,
-		Archive: p.Archive,
-		Slices:  make(map[string]yamlSlice, len(p.Slices)),
+// sliceToYAML converts a Slice object to a yamlSlice object.
+func sliceToYAML(s *Slice) (*yamlSlice, error) {
+	slice := &yamlSlice{
+		Contents:  make(map[string]*yamlPath, len(s.Contents)),
+		Mutate:    s.Scripts.Mutate,
+		Essential: make(map[string]*yamlEssential, len(s.Essential)),
 	}
-	for name, slice := range p.Slices {
-		yamlSlice, err := sliceToYAML(slice)
+	for key, info := range s.Essential {
+		slice.Essential[key.String()] = &yamlEssential{Arch: yamlArch{info.Arch}}
+	}
+	for path, info := range s.Contents {
+		yamlPath, err := pathInfoToYAML(&info)
 		if err != nil {
 			return nil, err
 		}
-		pkg.Slices[name] = *yamlSlice
+		slice.Contents[path] = yamlPath
 	}
-	return pkg, nil
+	return slice, nil
+}
+
+type v1v2YAMLSlice struct {
```

Comment:
If/when we do need such types and functions, it's better to spell this as `yamlSliceV2` and `yamlSliceV3`, instead of `v1v2YAMLSlice` and `yamlSlice`. Same for functions, etc.